VeganVirus/VeganVirus/Animation.cpp

```diff
@@ -0,0 +1,54 @@
+#include "Animation.h"
+
+Animation::Animation(Draw* draw, double timeBetweenFrames, const std::vector<const wchar_t*> frameImagePaths)
+{
+	_draw = draw;
+	_timeBetweenFrames = timeBetweenFrames;
+	this->reset();
+
+	// load bitmaps
+	for (const wchar_t* framePath : frameImagePaths)
+	{
+		this->_bitmaps.push_back(new Bitmap(framePath));
+	}
+}
+
+Animation::Animation(Draw* draw, double timeBetweenFrames, const std::vector<const wchar_t*> frameImagePaths, double w, double h)
+{
+	_draw = draw;
+	_timeBetweenFrames = timeBetweenFrames;
+	this->reset();
+
+	// load bitmaps
+	for (const wchar_t* framePath : frameImagePaths)
```

Comment:
please define the variable `framePath` at the start of the function.

Reply:
In for and foreach loops, the variable is usually defined inside the loop :)

Reply:
alon is right ariel.